dm_control/locomotion/tasks/reference_pose/rewards.py

```diff
@@ -23,27 +23,27 @@
                                         ['reward', 'debug', 'reward_terms'])
 
 
-def bounded_quat_dist(source, target):
-  """Computes a quaternion distance limiting the difference to a max of pi.
+def bounded_quat_dist(source: np.ndarray,
+                      target: np.ndarray) -> np.ndarray:
+  """Computes a quaternion distance limiting the difference to a max of pi/2.
 
-  This is achieved by taking the minimum of d(source, target) and
-  d(source, -target).
-
-  This function supports batched inputs.
+  This function supports an arbitrary number of batch dimensions, B.
 
   Args:
-    source: a quaternion
-    target: another quaternion
+    source: a quaternion, shape (B, 4).
+    target: another quaternion, shape (B, 4).
 
   Returns:
-    quaternion distance.
+    Quaternion distance, shape (B, 1).
   """
   source /= np.linalg.norm(source, axis=-1, keepdims=True)
   target /= np.linalg.norm(target, axis=-1, keepdims=True)
-  default_dist = tr.quat_dist(source, target)
```

Comment:
Please could you remove the `from dm_control.utils import transformations as tr` import as it's no longer used

Reply:
I've just added all the changes: removed old import, switched to einsum, and improved style.